extract.py

```diff
@@ -41,17 +121,30 @@ def get_calls(child):
 def get_tus(index, path):
     for file in os.listdir(path):
         if file.endswith(".c"):
-            yield index.parse(path + "/" + file)
+            split_line = 0
+            # this works on the assumption that all includes are at the start of the file.
+            # calculating the split line gives information about which things are defined in
+            # headers and which are defined in in the c file.
+            with open(path + "/" + file, "r") as f:
+                for current_line_no, line in enumerate(f):
+                    match = re.match("# [0-9]+ \"" + file + "\" 2", line)
```

Comment:
Is the `[0-9]+` to work with generated `#`s from the preprocessor?

Reply:
Yes
```c
# 34 "ldo.c" 2
# 84 "ldo.c"
```
This is how they look like, and can be used to identify the different portions of code that comprise the final source file.

Reply:
Got it!